build.zig

```diff
@@ -0,0 +1,459 @@
+        } else {
+            flags.append("-D GC_USE_DLOPEN_WRAP") catch unreachable;
+        }
+    }
+
+    if (enable_munmap) {
+        flags.append("-D USE_MMAP") catch unreachable;
+        flags.append("-D USE_MUNMAP") catch unreachable;
+    } else if (enable_mmap) {
+        flags.append("-D USE_MMAP") catch unreachable;
+    }
+
+    if (!enable_dynamic_loading) {
+        flags.append("-D IGNORE_DDYNAMIC_LOADING") catch unreachable;
+    }
+
+    if (!enable_register_main_static_data) {
+        flags.append("-D GC_DONT_REGISTER_MAIN_STATIC_DATA") catch unreachable;
+    }
+
+
+    if (enable_large_config) {
+        flags.append("-D LARGE_CONFIG") catch unreachable;
+    }
+
+    if (enable_gc_assertions) {
+        flags.append("-D GC_ASSERTIONS") catch unreachable;
+    }
+
+    if (!enable_threads_discovery) {
+        flags.append("-D GC_NO_THREADS_DISCOVERY") catch unreachable;
+    }
+
+    if (enable_rwlock) {
+        flags.append("-D USE_RWLOCK") catch unreachable;
+    }
+
+    if (enable_checksums) {
+        if (enable_munmap or enable_threads) {
+            @panic("CHECKSUMS not compatible with USE_MUNMAP or threads");
+        }
+        flags.append("-D CHECKSUMS") catch unreachable;
+        source_files.appendSlice(&.{
+            "checksums.c"
+        }) catch unreachable;
+    }
+
+    if (build_shared_libs) {
+        source_files.clearAndFree();
+        source_files.appendSlice(&.{
+            "extra/gc.c"
+        }) catch unreachable;
+        if (enable_threads and !(target.isDarwin() or target.isWindows())) {
+            flags.append("-D GC_PTHREAD_START_STANDALONE") catch unreachable;
+            source_files.appendSlice(&.{
+                "pthread_start.c",
+            }) catch unreachable;
+        }
+    }
+
+    if (target.isDarwin()) {
+        flags.append("-D HAVE_PTHREAD_SETNAME_NP_WITHOUT_TID") catch unreachable;
+    } else if (target.isLinux()) {
+        flags.append("-D HAVE_PTHREAD_SETNAME_NP_WITH_TID") catch unreachable;
+    } else {
+        // TODO: hmm, this probably needs more conditions on more platforms!?
+        flags.append("-D HAVE_PTHREAD_SETNAME_NP_WITH_TID_AND_ARG") catch unreachable;
+    }
+
+    // Zig uses clang which ships with these so unless another sysroot / libc
+    // etc headers is pointed out, it is fine to hard-code enable this.
+    flags.append("-D HAVE_DLADDR") catch unreachable;
+    flags.append("-D HAVE_DL_ITERATE_PHDR") catch unreachable;
+    flags.append("-D HAVE_SYS_TYPES_H") catch unreachable;
+    flags.append("-D HAVE_UNISTD_H") catch unreachable;
+    // We HAVE_WCSLEN, so enabling..
+    flags.append("-D GC_REQUIRE_WCSDUP") catch unreachable;
+
+    if (optimize == .Debug) {
+        flags.append("-D NDEBUG") catch unreachable;
+    }
+
+    if (build_shared_libs) {
+        flags.append("-D GC_DLL") catch unreachable;
+// TODO: what to do about visibility=hidden? is it good or bad?
```

Comment:
I think clang supports it everywhere (it might be meaningless on some targets, like Windows DLL but should not be harmful)